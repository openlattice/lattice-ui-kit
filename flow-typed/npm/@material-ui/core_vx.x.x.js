// flow-typed signature: 7ed29ef6da93a10ffdf64ee435c9faae
// flow-typed version: <<STUB>>/@material-ui/core_v4.5.2/flow_v0.117.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@material-ui/core'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@material-ui/core' {
  declare module.exports: any;
}

declare module '@material-ui/core/styles' {
  declare module.exports: any;
}
