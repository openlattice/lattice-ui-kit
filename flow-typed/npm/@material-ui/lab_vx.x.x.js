// flow-typed signature: a13bdd1d57cdd924624da6dae303f597
// flow-typed version: <<STUB>>/@material-ui/lab_v4.0.0-alpha.40/flow_v0.117.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@material-ui/lab'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@material-ui/lab' {
  declare module.exports: any;
}
