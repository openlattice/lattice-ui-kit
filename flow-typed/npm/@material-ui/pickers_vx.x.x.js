// flow-typed signature: 961daa4d7c37688be7a4da00b615a7ba
// flow-typed version: <<STUB>>/@material-ui/pickers_v~3.2.10/flow_v0.123.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@material-ui/pickers'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@material-ui/pickers' {
  declare module.exports: any;
}
