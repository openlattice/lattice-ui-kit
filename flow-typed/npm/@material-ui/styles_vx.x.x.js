// flow-typed signature: 5f9d61d50037b50ea5904b8abd1b18ff
// flow-typed version: <<STUB>>/@material-ui/styles_v~4.9.0/flow_v0.123.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@material-ui/styles'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@material-ui/styles' {
  declare module.exports: any;
}
