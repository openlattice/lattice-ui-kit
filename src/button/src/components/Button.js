/*
 * @flow
 */

import React, { Component } from 'react';
import type { Node } from 'react';

import PropTypes from 'prop-types';

import DefaultButton from './styled/DefaultButton';
import PrimaryButton from './styled/PrimaryButton';
import SecondaryButton from './styled/SecondaryButton';

type ButtonMode =
  | 'default'
  | 'primary'
  | 'secondary';

type Props = {
  children :Node;
  className ? :string;
  disabled ? :boolean;
  mode ? :ButtonMode;
  onClick :(e :SyntheticEvent<HTMLButtonElement>) => void;
  type ? :string;
};

type State = {};

/*
 * Inspiration:
 * https://atlaskit.atlassian.com/packages/core/button
 * https://evergreen.surge.sh/components/buttons
 */
export default class Button extends Component<Props, State> {

  static propTypes = {
    children: PropTypes.node.isRequired,
    className: PropTypes.string,
    disabled: PropTypes.bool,
    mode: PropTypes.string,
    onClick: PropTypes.func.isRequired,
    type: PropTypes.string,
  }

  static defaultProps = {
    className: '',
    disabled: false,
    mode: 'default',
    type: 'button',
  }

  render() {

    const {
      children,
      className,
      disabled,
      mode,
      onClick,
      type,
    } = this.props;

    switch (mode) {
      case 'primary':
        return (
          <PrimaryButton className={className} disabled={disabled} onClick={onClick} type={type}>
            { children }
          </PrimaryButton>
        );
      case 'secondary':
        return (
          <SecondaryButton className={className} disabled={disabled} onClick={onClick} type={type}>
            { children }
          </SecondaryButton>
        );
      default:
        return (
          <DefaultButton className={className} disabled={disabled} onClick={onClick} type={type}>
            { children }
          </DefaultButton>
        );
    }
  }
}
