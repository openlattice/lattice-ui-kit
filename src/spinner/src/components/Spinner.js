import styled, { keyframes } from 'styled-components';

const spin = keyframes`
  0% {
    transform: rotate(0);
  }

  100% {
    transform: rotate(360deg);
  }
`;

const Spinner = styled.div`
  &::before {
    animation: ${spin} 0.75s linear infinite;
    border: 1px solid #c5d5e5;
    border-radius: 50%;
    border-top-color: #135;
    box-sizing: border-box;
    content: '';
    top: 50%;
    left: 50%;
    width: 50px;
    height: 50px;
    margin-top: -25px; /* half the height */
    margin-left: -25px; /* half the width */
    position: absolute;
  }
`;

export default Spinner;
