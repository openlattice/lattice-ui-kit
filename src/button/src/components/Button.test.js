import React from 'react';

import capitalize from 'lodash/capitalize';
import toJson from 'enzyme-to-json';
import { Button as MuiButton } from '@material-ui/core';
import { createMount } from '@material-ui/core/test-utils';

import Button from './Button';

import Spinner from '../../../spinner';
import { nope } from '../../../utils/testing/MockUtils';

const DEFAULT_BTN_TXT = 'DEFAULT_BUTTON';

const variants = ['contained', 'outlined', 'text'];
const themedColors = ['primary', 'secondary'];
const intentColors = [
  'error',
  'info',
  'success',
  'warning',
];
const sizes = ['small', 'large'];

describe('button', () => {

  let mount;

  beforeAll(() => {
    mount = createMount();
  });

  afterAll(() => {
    mount.cleanUp();
  });

  describe('props', () => {
    test('fullWidth should add MuiButton-fullWidth class', () => {
      const wrapper = mount(<Button fullWidth>{ DEFAULT_BTN_TXT }</Button>);

      const button = wrapper.find('button.MuiButton-fullWidth');
      expect(button).toHaveLength(1);
    });

    test('disabled should add Mui-disabled class', () => {
      const wrapper = mount(<Button disabled>{ DEFAULT_BTN_TXT }</Button>);

      const button = wrapper.find('button.Mui-disabled');
      expect(button).toHaveLength(1);
    });

    test('isLoading should set startIcon to Spinner', () => {
      const wrapper = mount(<Button isLoading onClick={nope}>{ DEFAULT_BTN_TXT }</Button>);

      const disabled = wrapper.find('button.Mui-disabled');
      expect(disabled).toHaveLength(1);

      const span = wrapper.find('span.MuiButton-startIcon');
      expect(span).toHaveLength(1);

      const button = wrapper.find(MuiButton);
      expect(button.prop('startIcon').type).toEqual(<Spinner />.type);
    });
  });

  describe('variants & colors', () => {

    test('render contained by default', () => {
      const wrapper = mount(<Button>{ DEFAULT_BTN_TXT }</Button>);

      const button = wrapper.find('button.MuiButton-contained');
      expect(button).toHaveLength(1);
    });

    test('should each render with size', () => {
      variants.forEach((variant) => {
        // default color
        const wrapper = mount(<Button variant={variant}>{ DEFAULT_BTN_TXT }</Button>);
        const button = wrapper.find('button');
        expect(button.hasClass(`MuiButton-${variant}`)).toEqual(true);
        expect(toJson(wrapper)).toMatchSnapshot();

        // sizes
        sizes.forEach((size) => {
          const sizeWrapper = mount(<Button variant={variant} size={size}>{ DEFAULT_BTN_TXT }</Button>);
          const sizeButton = sizeWrapper.find('button');
          expect(sizeButton.hasClass(`MuiButton-size${capitalize(size)}`)).toEqual(true);
          expect(sizeButton.hasClass(`MuiButton-${variant}Size${capitalize(size)}`)).toEqual(true);
          expect(toJson(sizeWrapper)).toMatchSnapshot();
        });

        // themed colors
        themedColors.forEach((color) => {
          const themedColorWrapper = mount(<Button variant={variant} color={color}>{ DEFAULT_BTN_TXT }</Button>);
          const themedColorButton = themedColorWrapper.find('button');
          expect(themedColorButton.hasClass(`MuiButton-${variant}${capitalize(color)}`)).toEqual(true);
          expect(toJson(themedColorWrapper)).toMatchSnapshot();

          sizes.forEach((size) => {
            const sizeWrapper = mount(<Button variant={variant} color={color} size={size}>{ DEFAULT_BTN_TXT }</Button>);
            const sizeButton = sizeWrapper.find('button');
            expect(sizeButton.hasClass(`MuiButton-size${capitalize(size)}`)).toEqual(true);
            expect(sizeButton.hasClass(`MuiButton-${variant}Size${capitalize(size)}`)).toEqual(true);
            expect(toJson(sizeWrapper)).toMatchSnapshot();
          });
        });

        // intent colors
        intentColors.forEach((color) => {
          const intentColorWrapper = mount(<Button variant={variant} color={color}>{ DEFAULT_BTN_TXT }</Button>);
          const intentColorButton = intentColorWrapper.find('button');
          const regex = new RegExp(`makeStyles-${variant}${capitalize(color)}-\\d+`);
          expect(intentColorButton.hasClass(regex)).toEqual(true);
          expect(toJson(intentColorWrapper)).toMatchSnapshot();

          sizes.forEach((size) => {
            const sizeWrapper = mount(<Button variant={variant} color={color} size={size}>{ DEFAULT_BTN_TXT }</Button>);
            const sizeButton = sizeWrapper.find('button');
            expect(sizeButton.hasClass(`MuiButton-size${capitalize(size)}`)).toEqual(true);
            expect(sizeButton.hasClass(`MuiButton-${variant}Size${capitalize(size)}`)).toEqual(true);
            expect(toJson(sizeWrapper)).toMatchSnapshot();
          });
        });
      });
    });
  });

});
