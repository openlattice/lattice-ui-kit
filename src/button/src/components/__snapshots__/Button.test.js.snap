// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button mode="default" should match snapshot 1`] = `
.c1 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  max-width: 100%;
  position: relative;
  overflow: hidden;
  min-width: 14px;
}

.c2 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  opacity: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c0 {
  background-color: #f0f0f7;
  border-color: #f0f0f7;
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  color: #8e929b;
  cursor: pointer;
  font-size: 14px;
  line-height: 18px;
  outline: none;
  padding: 10px 20px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
}

.c0:hover {
  background-color: #dcdce7;
  border-color: #dcdce7;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:active {
  background-color: #b6bbc7;
  border-color: #b6bbc7;
  color: #ffffff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:disabled {
  background-color: #f0f0f7;
  border-color: #f0f0f7;
  color: #b6bbc7;
  cursor: not-allowed;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<Button
  disabled={false}
  isLoading={false}
  mode="default"
  onClick={[Function]}
  type="button"
>
  <DefaultButton
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <StyledComponent
      disabled={false}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "DefaultButton-sc-2dhy0g-0",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "background-color:",
              "#f0f0f7",
              ";border-color:",
              "#f0f0f7",
              ";border-radius:4px;border-style:solid;border-width:1px;color:",
              "#8e929b",
              ";cursor:pointer;font-size:14px;line-height:18px;outline:none;padding:10px 20px;text-align:center;text-decoration:none;white-space:nowrap;",
              " ",
              "&:hover{background-color:",
              "#dcdce7",
              ";border-color:",
              "#dcdce7",
              ";cursor:pointer;text-decoration:none;}",
              " ",
              "&:active{background-color:",
              "#b6bbc7",
              ";border-color:",
              "#b6bbc7",
              ";color:",
              "#ffffff",
              ";text-decoration:none;}",
              " ",
              "&:disabled{background-color:",
              "#f0f0f7",
              ";border-color:",
              "#f0f0f7",
              ";color:",
              "#b6bbc7",
              ";cursor:not-allowed;text-decoration:none;}",
            ],
          },
          "displayName": "DefaultButton",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "DefaultButton-sc-2dhy0g-0",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
      type="button"
    >
      <button
        className="c0"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <ContentWrapper>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "ContentWrapper-sc-36hfub-0",
                  "isStatic": true,
                  "lastClassName": "c1",
                  "rules": Array [
                    "align-self:center;display:inline-flex;flex-wrap:nowrap;max-width:100%;position:relative;overflow:hidden;min-width:14px;",
                  ],
                },
                "displayName": "ContentWrapper",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "ContentWrapper-sc-36hfub-0",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="c1"
            >
              <Content
                isLoading={false}
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "Content-sc-18psqsj-0",
                        "isStatic": false,
                        "lastClassName": "c2",
                        "rules": Array [
                          "flex:1 1 auto;opacity:",
                          [Function],
                          ";overflow:hidden;text-overflow:ellipsis;white-space:nowrap;",
                        ],
                      },
                      "displayName": "Content",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "Content-sc-18psqsj-0",
                      "target": "span",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  isLoading={false}
                >
                  <span
                    className="c2"
                  >
                    DEFAULT_BUTTON
                  </span>
                </StyledComponent>
              </Content>
            </span>
          </StyledComponent>
        </ContentWrapper>
      </button>
    </StyledComponent>
  </DefaultButton>
</Button>
`;

exports[`button mode="default" should match snapshot 2`] = `
.c1 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  max-width: 100%;
  position: relative;
  overflow: hidden;
  min-width: 14px;
}

.c2 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  opacity: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c0 {
  background-color: #f0f0f7;
  border-color: #f0f0f7;
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  color: #8e929b;
  cursor: pointer;
  font-size: 14px;
  line-height: 18px;
  outline: none;
  padding: 10px 20px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
}

.c0:hover {
  background-color: #dcdce7;
  border-color: #dcdce7;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:active {
  background-color: #b6bbc7;
  border-color: #b6bbc7;
  color: #ffffff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:disabled {
  background-color: #f0f0f7;
  border-color: #f0f0f7;
  color: #b6bbc7;
  cursor: not-allowed;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<Button
  disabled={false}
  isLoading={false}
  mode="default"
  onClick={[Function]}
  type="button"
>
  <DefaultButton
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <StyledComponent
      disabled={false}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "DefaultButton-sc-2dhy0g-0",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "background-color:",
              "#f0f0f7",
              ";border-color:",
              "#f0f0f7",
              ";border-radius:4px;border-style:solid;border-width:1px;color:",
              "#8e929b",
              ";cursor:pointer;font-size:14px;line-height:18px;outline:none;padding:10px 20px;text-align:center;text-decoration:none;white-space:nowrap;",
              " ",
              "&:hover{background-color:",
              "#dcdce7",
              ";border-color:",
              "#dcdce7",
              ";cursor:pointer;text-decoration:none;}",
              " ",
              "&:active{background-color:",
              "#b6bbc7",
              ";border-color:",
              "#b6bbc7",
              ";color:",
              "#ffffff",
              ";text-decoration:none;}",
              " ",
              "&:disabled{background-color:",
              "#f0f0f7",
              ";border-color:",
              "#f0f0f7",
              ";color:",
              "#b6bbc7",
              ";cursor:not-allowed;text-decoration:none;}",
            ],
          },
          "displayName": "DefaultButton",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "DefaultButton-sc-2dhy0g-0",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
      type="button"
    >
      <button
        className="c0"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <ContentWrapper>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "ContentWrapper-sc-36hfub-0",
                  "isStatic": true,
                  "lastClassName": "c1",
                  "rules": Array [
                    "align-self:center;display:inline-flex;flex-wrap:nowrap;max-width:100%;position:relative;overflow:hidden;min-width:14px;",
                  ],
                },
                "displayName": "ContentWrapper",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "ContentWrapper-sc-36hfub-0",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="c1"
            >
              <Content
                isLoading={false}
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "Content-sc-18psqsj-0",
                        "isStatic": false,
                        "lastClassName": "c2",
                        "rules": Array [
                          "flex:1 1 auto;opacity:",
                          [Function],
                          ";overflow:hidden;text-overflow:ellipsis;white-space:nowrap;",
                        ],
                      },
                      "displayName": "Content",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "Content-sc-18psqsj-0",
                      "target": "span",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  isLoading={false}
                >
                  <span
                    className="c2"
                  >
                    DEFAULT_BUTTON
                  </span>
                </StyledComponent>
              </Content>
            </span>
          </StyledComponent>
        </ContentWrapper>
      </button>
    </StyledComponent>
  </DefaultButton>
</Button>
`;

exports[`button mode="primary" should match snapshot 1`] = `
.c1 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  max-width: 100%;
  position: relative;
  overflow: hidden;
  min-width: 14px;
}

.c2 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  opacity: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c0 {
  background-color: #8045ff;
  border-color: #8045ff;
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  color: #ffffff;
  cursor: pointer;
  font-size: 14px;
  line-height: 18px;
  outline: none;
  padding: 10px 20px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
}

.c0:hover {
  background-color: #6124e2;
  border-color: #6124e2;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:active {
  background-color: #361876;
  border-color: #361876;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:disabled {
  background-color: #f0f0f7;
  border-color: #f0f0f7;
  color: #b6bbc7;
  cursor: not-allowed;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<Button
  disabled={false}
  isLoading={false}
  mode="primary"
  onClick={[Function]}
  type="button"
>
  <PrimaryButton
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <StyledComponent
      disabled={false}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "PrimaryButton-sc-1t8zuv9-0",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "background-color:",
              "#8045ff",
              ";border-color:",
              "#8045ff",
              ";border-radius:4px;border-style:solid;border-width:1px;color:",
              "#ffffff",
              ";cursor:pointer;font-size:14px;line-height:18px;outline:none;padding:10px 20px;text-align:center;text-decoration:none;white-space:nowrap;",
              " ",
              "&:hover{background-color:",
              "#6124e2",
              ";border-color:",
              "#6124e2",
              ";cursor:pointer;text-decoration:none;}",
              " ",
              "&:active{background-color:",
              "#361876",
              ";border-color:",
              "#361876",
              ";text-decoration:none;}",
              " ",
              "&:disabled{background-color:",
              "#f0f0f7",
              ";border-color:",
              "#f0f0f7",
              ";color:",
              "#b6bbc7",
              ";cursor:not-allowed;text-decoration:none;}",
            ],
          },
          "displayName": "PrimaryButton",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "PrimaryButton-sc-1t8zuv9-0",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
      type="button"
    >
      <button
        className="c0"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <ContentWrapper>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "ContentWrapper-sc-36hfub-0",
                  "isStatic": true,
                  "lastClassName": "c1",
                  "rules": Array [
                    "align-self:center;display:inline-flex;flex-wrap:nowrap;max-width:100%;position:relative;overflow:hidden;min-width:14px;",
                  ],
                },
                "displayName": "ContentWrapper",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "ContentWrapper-sc-36hfub-0",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="c1"
            >
              <Content
                isLoading={false}
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "Content-sc-18psqsj-0",
                        "isStatic": false,
                        "lastClassName": "iDvfiU",
                        "rules": Array [
                          "flex:1 1 auto;opacity:",
                          [Function],
                          ";overflow:hidden;text-overflow:ellipsis;white-space:nowrap;",
                        ],
                      },
                      "displayName": "Content",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "Content-sc-18psqsj-0",
                      "target": "span",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  isLoading={false}
                >
                  <span
                    className="c2"
                  >
                    PRIMARY_BUTTON
                  </span>
                </StyledComponent>
              </Content>
            </span>
          </StyledComponent>
        </ContentWrapper>
      </button>
    </StyledComponent>
  </PrimaryButton>
</Button>
`;

exports[`button mode="secondary" should match snapshot 1`] = `
.c1 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  max-width: 100%;
  position: relative;
  overflow: hidden;
  min-width: 14px;
}

.c2 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  opacity: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c0 {
  background-color: #e4d8ff;
  border-color: #e4d8ff;
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  color: #6124e2;
  cursor: pointer;
  font-size: 14px;
  line-height: 18px;
  outline: none;
  padding: 10px 20px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
}

.c0:hover {
  background-color: #d0bbff;
  border-color: #d0bbff;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:active {
  background-color: #b898ff;
  border-color: #b898ff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:disabled {
  background-color: #f0f0f7;
  border-color: #f0f0f7;
  color: #b6bbc7;
  cursor: not-allowed;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<Button
  disabled={false}
  isLoading={false}
  mode="secondary"
  onClick={[Function]}
  type="button"
>
  <SecondaryButton
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <StyledComponent
      disabled={false}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "SecondaryButton-s3wp95-0",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "background-color:",
              "#e4d8ff",
              ";border-color:",
              "#e4d8ff",
              ";border-radius:4px;border-style:solid;border-width:1px;color:",
              "#6124e2",
              ";cursor:pointer;font-size:14px;line-height:18px;outline:none;padding:10px 20px;text-align:center;text-decoration:none;white-space:nowrap;",
              " ",
              "&:hover{background-color:",
              "#d0bbff",
              ";border-color:",
              "#d0bbff",
              ";cursor:pointer;text-decoration:none;}",
              " ",
              "&:active{background-color:",
              "#b898ff",
              ";border-color:",
              "#b898ff",
              ";text-decoration:none;}",
              " ",
              "&:disabled{background-color:",
              "#f0f0f7",
              ";border-color:",
              "#f0f0f7",
              ";color:",
              "#b6bbc7",
              ";cursor:not-allowed;text-decoration:none;}",
            ],
          },
          "displayName": "SecondaryButton",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "SecondaryButton-s3wp95-0",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
      type="button"
    >
      <button
        className="c0"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <ContentWrapper>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "ContentWrapper-sc-36hfub-0",
                  "isStatic": true,
                  "lastClassName": "c1",
                  "rules": Array [
                    "align-self:center;display:inline-flex;flex-wrap:nowrap;max-width:100%;position:relative;overflow:hidden;min-width:14px;",
                  ],
                },
                "displayName": "ContentWrapper",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "ContentWrapper-sc-36hfub-0",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="c1"
            >
              <Content
                isLoading={false}
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "Content-sc-18psqsj-0",
                        "isStatic": false,
                        "lastClassName": "c2",
                        "rules": Array [
                          "flex:1 1 auto;opacity:",
                          [Function],
                          ";overflow:hidden;text-overflow:ellipsis;white-space:nowrap;",
                        ],
                      },
                      "displayName": "Content",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "Content-sc-18psqsj-0",
                      "target": "span",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  isLoading={false}
                >
                  <span
                    className="c2"
                  >
                    SECONDARY_BUTTON
                  </span>
                </StyledComponent>
              </Content>
            </span>
          </StyledComponent>
        </ContentWrapper>
      </button>
    </StyledComponent>
  </SecondaryButton>
</Button>
`;
